adapters/slack/usergroup/usergroup.go

```diff
@@ -183,25 +186,31 @@ func (u *UserGroup) Add(ctx context.Context, emails []string) error {
 			return fmt.Errorf("slack.usergroup.add.getuserbyemail(%s) -> %w", email, err)
 		}
 
-		// Add the new email user IDs to the list.
-		updatedUserGroup = append(updatedUserGroup, user.ID)
+		if !(slices.Contains(updatedUserGroup, user.ID)) {
+			// Add the new email user IDs to the list.
+			updatedUserGroup = append(updatedUserGroup, user.ID)
 
-		// Add the new email user ID to the cache
-		u.cache[email] = user.ID
+			// Add the new email user ID to the cache
+			u.cache[email] = user.ID
+		}
 
 		// Calls to GetUserByEmail are heavily rate limited, so sleep to avoid this.
 		time.Sleep(2 * time.Second) //nolint:gomnd
 	}
 
-	// Add the members to the Slack UserGroup.
-	joinedSlackIds := strings.Join(updatedUserGroup, ",")
+	if reflect.DeepEqual(updatedUserGroup, initialUserGroup) {
```

Comment:
I know this is a relatively expensive operation. Happy to refactor to use a var to track when the Contains test succeeds if that would be preferable but I thought this was more readable